react/features/filmstrip/components/native/TileView.js

```diff
@@ -0,0 +1,324 @@
+
+import {
+    getNearestReceiverVideoQualityLevel,
+    setMaxReceiverVideoQuality
+} from '../../../base/conference';
+import {
+    DimensionsDetector,
+    isNarrowAspectRatio,
+    makeAspectRatioAware
+} from '../../../base/responsive-ui';
+
+import Thumbnail from './Thumbnail';
+import styles from './styles';
+
+/**
+ * The type of the React {@link Component} props of {@link TileView}.
+ */
+type Props = {
+
+    /**
+     * The participants in the conference.
+     */
+    _participants: Array<any>,
+
+    /**
+     * Invoked to update the receiver video quality.
+     */
+    dispatch: Dispatch<*>,
+
+    /**
+     * Callback to invoke when tile view is tapped.
+     */
+    onClick: Function
+};
+
+/**
+ * The type of the React {@link Component} state of {@link TileView}.
+ */
+type State = {
+
+    /**
+     * The available width for {@link TileView} to occupy.
+     */
+    height: number,
+
+    /**
+     * The available height for {@link TileView} to occupy.
+     */
+    width: number
+};
+
+const TILE_ASPECT_RATIO = 1;
+
+/**
+ * Implements a React {@link Component} which represents the filmstrip on
+ * mobile/React Native.
+ *
+ * @extends Component
+ */
+class TileView extends Component<Props, State> {
+    state = {
+        height: 0,
+        width: 0
+    };
+
+    /**
+     * Initializes a new {@code TileView} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props: Props) {
+        super(props);
+
+        // Bind event handler so it is only bound once per instance.
+        this._onDimensionsChanged = this._onDimensionsChanged.bind(this);
+    }
+
+    /**
+     * Updates the receiver video quality.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        this._updateReceiverQuality();
+    }
+
+    /**
+     * Updates the receiver video quality.
+     *
+     * @inheritdoc
+     */
+    componentDidUpdate() {
+        this._updateReceiverQuality();
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { onClick } = this.props;
+        const { height, width } = this.state;
+        const rowElements = this._groupIntoRows(
+            this._renderThumbnails(), this._getColumnCount());
+
+        return (
+            <DimensionsDetector
+                onDimensionsChanged = { this._onDimensionsChanged }>
+                <ScrollView
+                    style = {{
+                        ...styles.tileView,
+                        height,
+                        width
+                    }}>
+                    <TouchableWithoutFeedback onPress = { onClick }>
+                        <View
+                            style = {{
+                                ...styles.tileViewRows,
+                                minHeight: height,
```

Comment:
Do these style props do anything here?

Reply:
Yes they do stuff; I tested them again just now to make sure. `style.tileViewRows` is used to make sure the tiles are vertical aligned center on the screen. minHeight and minWidth are used to make sure the container takes up at least the full screen; that way when there are only a few thumbnails, the thumbnails will display in the middle of the screen thanks to center alignment. minX is used so that the container is allowed to grow as needed, beyond the size of the parent container, so that thumbnails can become scrollable. Arguably minWidth isn't really needed because there should be no horizontal scrolling. So that's my answer and I don't claim it to be the right way but was the culmination of me playing around for a while.

Reply:
Hm, interesting, I didn't know it would work like this. Okay, thanks!